### PR TITLE
Mark intrusive_graph_node::has_[dependency|dependent] as const

### PR DESCRIPTION
**Pull request by psalz**
10chars

### CODE REVIEW

File: include/intrusive_graph.h [hunk 1]
```diff
@@ -151,6 +151,13 @@ namespace detail {
 			if(it == deps.end()) return std::nullopt;
 			return it;
 		}
+
+		template <typename Dep>
+		std::optional<typename gch::small_vector<Dep>::const_iterator> maybe_get_dep(const gch::small_vector<Dep>& deps, const T* node) const {
```
Comment (fknorr) [inline include/intrusive_graph.h:156]: That's essentially a static method, so there should be no need for const/non-const versions:

```c++
		template <typename Dep>
		static auto maybe_get_dep(gch::small_vector<Dep>& deps, T* node) {
			auto it = std::find_if(deps.begin(), deps.end(), [&](auto d) { return d.node == node; });
			if(it == deps.end()) return std::nullopt; // not sure if this typechecks
			return std::optional{it};
		}
```

**Review by fknorr — COMMENTED** [review]
The `const` qualifiers LGTM of course, but I can't help but scratch my head at the entire `maybe_get_dep` function. Why have an `optional<iterator>` when iterators are already nullable through `range.end()`?

If I were to design this from scratch it would look like so:

```c++
public:
    bool has_dependency(T* node, std::optional<dependency_kind> kind = std::nullopt) const {
        if (auto it = find_by_node(m_dependencies, node); it != m_dependencies.end()) {
            ...
        }
    }

private:
    template<typename Range>
    static auto find_by_node(Range &rng, T *const node) {
        using std::begin, std::end;
        return std::find_if(begin(rng), end(rng), [=](auto &d) { return d.node == node });
    }
```

Maybe this PR is the right place for a small refactoring.

**Comment by psalz** [comment]
You are absolutely right! Refactored.

**Review by fknorr — APPROVED** [review]
LGTM, although there are a bunch of variables that could be const (multiple iterators `it` / `this_it`, and function parameters to `has_dependency` / `has_dependent`).

**Comment by psalz** [comment]
Rebased onto current master, made everything const.

**Pull request merged**